src/screens/courses/CardScreen/index.tsx

```diff
@@ -24,26 +25,26 @@ const CardScreen = ({ index, exitConfirmationModal, setExitConfirmationModal, go
   const [isLeftSwipeEnabled, setIsLeftSwipeEnabled] = useState<boolean>(true);
   const [isRightSwipeEnabled, setIsRightSwipeEnabled] = useState<boolean>(false);
 
-  const onSwipe = (cardIndex: number, event) => {
-    if (event.nativeEvent.translationX > SWIPE_SENSIBILITY && cardIndex > 0 && isLeftSwipeEnabled) {
-      navigation.navigate(`card-${cardIndex - 1}`);
+  const onSwipe = Gesture.Pan().onEnd((event) => {
+    if (event.translationX > SWIPE_SENSIBILITY && index > 0 && isLeftSwipeEnabled) {
+      runOnJS(navigation.navigate)(`card-${index - 1}`);
     }
 
-    if (event.nativeEvent.translationX < -SWIPE_SENSIBILITY && isRightSwipeEnabled) {
-      navigation.navigate(`card-${cardIndex + 1}`);
+    if (event.translationX < -SWIPE_SENSIBILITY && isRightSwipeEnabled) {
+      runOnJS(navigation.navigate)(`card-${index + 1}`);
     }
-  };
+  });
 
   return (
-    <PanGestureHandler onGestureEvent={event => onSwipe(index, event)}
+    <GestureDetector gesture={onSwipe}
```

Comment:
avec SDK 49 on passe à Reanimated 3 qui déprécie l'utilisation de Reanimated 1 qui est utilisé pour Gesture Handler 1 (qu'on utilise avec PanGestureHandler) et donc notre utilisation de Gesture Handler ne fonctionne plus il faut utiliser la version 2 de gesture Handler (https://docs.swmansion.com/react-native-gesture-handler/docs/#rngh-20). 
Dans cette version 2 les différents handler sont remplacés par un unique GestureDetector et c'est ensuite dans la fonction qu'on appelle (onSwipe) qu'on détermine quel type de geste on fait (dans notre cas le Pan) (https://docs.swmansion.com/react-native-gesture-handler/docs/gestures/gesture). J'ai dû aussi modifié l'appelle à navigation car j'avais l'erreur suivante (lire la première réponse https://github.com/software-mansion/react-native-reanimated/discussions/1571) que j'ai réglé en rajoutant runOnJS en m'aidant de cet article (https://medium.com/@taitasciore/handling-pan-and-scroll-gestures-simultaneously-and-gracefully-with-gesture-handler-2-reanimated-63f0d8f72d3c)